### ISSUE TITLE
Plugin Installer License Notice

### ISSUE THREAD

**Issue by Automate1** [opensourceautomation/Open-Source-Automation#159]
Since plugins can be written by different developers and plugins can use a different software license than the core of OSA, it would be nice to be able to display a custom license notice and accept button during the plugin install process.

**Comment by VaughnRupp** [comment]
I agree with this need, but it really complicates the other requests to do automatic plugin upgrades and remote plugin installs.

I wonder what the options are, like if we can put notice in the logs pointing to the EULA and just saying if they do not accept, then this plugin should be uninstalled, but let it run by default.

Or a Accept on the first instance downloaded, but not on updates.

**Comment by KHerron** [comment]
What if the Plugin installer ask the user "Do you want to automatically install future upgrades?" Then store their answer in the Plugin's Object as a property. So, If they say "Yes", then the installer can skip the License. But I do think it should be there during the initial install.

**Comment by VaughnRupp** [comment]
How about the Plugins have an EULA property that can hold the text for the EULA, and then a property on the plugin object that is called "I accept the EULA" that defaults to false and the OSA service won't run plugins where that property is set to false?

Then this will not impact the installation at all and only affect the plugin running?    This would help us automate the installs and keep that process streamlined and allow the EULA to be handled in the main WebUI with the other plugin settings?

I think we can get this done in v049 if we go this route and get this ancient issue closed.

**Comment by VaughnRupp** [comment]
If this does go in on 049, it will impact every plugin, all of them will have to have their install.sql upgraded, and end users will have to upgrade every plugin to get it to run those installs.   This will make v049 a major release that will take lots of testing and could cause pain to users if not done correctly. 

On the bright side though, if we can get all of these changes done before we release a public stable version, then we are less likely to put future users though that pain.   So I am going through the old tickets and trying to work as many major changes as possible to make testing of 049 more productive.
